controllers/asosecret_controller_test.go

```diff
@@ -54,6 +54,7 @@ func TestASOSecretReconcile(t *testing.T) {
 	defaultAzureCluster := getASOAzureCluster()
 	defaultAzureManagedControlPlane := getASOAzureManagedControlPlane()
 	defaultASOSecret := getASOSecret(defaultAzureCluster)
+	defaultClusterClusterIdentityType := "ServicePrincipal"
```

Comment:
nit
```suggestion
	defaultClusterClusterIdentityType := infrav1.ServicePrincipal
```

Reply:
You're right. This was just a quick fix in order to make lint happy. Adjusted it based on your suggestion.